versioned_docs/version-4.6/developers/applications/defining-roles.md

```diff
@@ -1,54 +1,65 @@
 ---
-title: Defining Roles
+title: Defining Application Roles
 ---
 
-In addition to [defining a database schema](./defining-schemas), you can also define roles in your application. Roles are a way to group permissions together and assign them to users as part of Harper's [role based access control](../security/users-and-roles). An application component may declare roles that should exist for the application in a roles configuration file. To use this, first specify your roles config file in the `config.yaml` in your application directory:
+# Defining Application Roles
+
+Applications are more than just tables and endpoints — they need access rules. Harper lets you define roles directly in your application so you can control who can do what, without leaving your codebase.
+
+Let’s walk through creating a role, assigning it, and seeing it in action.
+
+## Step 1: Declare a Role
+
+First, point Harper to a roles configuration file. Add this to your `config.yaml`:
 
 ```yaml
 roles:
   files: roles.yaml
 ```
 
-Now you can create a roles.yaml in your application directory:
+Then create a simple `roles.yaml` in your application directory. For example, here’s a role that can only read and insert data into the `Dog` table:
 
 ```yaml
-declared-role:
-  super_user: false # This is a boolean value that indicates if the role is a super user or not
-  # Now we can grant the permissions to databases, here we grant permissions to the default data database
-  data: # This is the same structure as role object that is used in the roles operations APIs
-    TableOne:
+dog_reader:
+  super_user: false
+  data:
+    Dog:
       read: true
       insert: true
-    TableTwo:
-      read: true
-      insert: false
-      update: true
-      delete: true
-      attributes:
-        name:
-          read: true
-          insert: false
-          update: true
 ```
 
-With this in place, where Harper starts up, it will create the roles in the roles.yaml file if they do not already exist. If they do exist, it will update the roles with the new permissions. This allows you to manage your roles in your application code and have them automatically created or updated when the application starts.
+When Harper starts up, it will create this role (or update it if it already exists).
+
+## Step 2: Assign the Role
+
+Now that the role exists, assign it to a user. You can do this through the [Users and Roles API](../security/users-and-roles) or via the CLI. Once assigned, the user’s permissions will reflect exactly what you declared in `roles.yaml`.
+
+For example, a user with the `dog_reader` role can insert new dog records, but not delete or update them.
```

Comment:
Need to walk the user through these steps now. Help them create a non-super-user user and assign that user this role. Then demonstrate some requests. Show how given one role or the other (in the authentication header of the request) results in different responses (likely 200 vs 4xx)